Source/Parsing/Frame/Content/Adapter/ID3TableOfContentsStringAdapter.swift

```diff
@@ -42,16 +48,20 @@ class ID3TableOfContentsStringAdapter: TableOfContentsStringAdapter {
         }
     }
 
-    private func getEntryCountFrom(contentComponents: [String]) -> Int {
-        return Int(contentComponents[3]) ?? 0
+    private func getEntryCountFrom(contentComponents: [String]) -> UInt8 {
+        #warning("Is falling back to zero really the right thing to do?")
+        #warning("Can 3 be out of bounds? Is trapping in this case okay?")
+        return UInt8(contentComponents[3]) ?? 0
     }
 
-    private func getChildElementIDsFrom(contentComponents: [String]) -> [String?] {
+    private func getChildElementIDsFrom(contentComponents: [String]) -> [String] {
+        #warning("Can 4 be out of bounds? Is trapping in this case okay?")
+        #warning("Why are we wrapping a single string in an array?")
```

Comment:
Ideally it shouldn't be a single string, but it's possible it could be. Like I said, if you're going to have a table of contents, it has to have at least 1 child element. But there could be hundreds of child elements (literally, I just listened to an audiobook with 150 chapters, and if it had been a chaptered MP3, that would have meant 150 child element IDs).

Again, this one shouldn't be out of bounds. The child-element IDs are mandatory for compliance, I believe?

Reply:
> But there could be hundreds of child elements

Not right now. The way it is written, there can only ever be one. But I suspected it should instead behave as you now describe, so I left the note to come back and sort it out.